lang/LangPrimSource/SC_CoreMIDI.cpp

```diff
@@ -719,24 +736,31 @@ static MIDITimeStamp midiTime(float latencySeconds) {
 
 #else
 
-static MIDITimeStamp midiTime(float latencySeconds) {
-    // add the latency expressed in seconds, to the current host time base.
-    UInt64 latencyNanos = 1000000000 * latencySeconds; // secs to nano
-    return (MIDITimeStamp)AudioGetCurrentHostTime() + AudioConvertNanosToHostTime(latencyNanos);
+static MIDITimeStamp midiTime(float latencySeconds, UInt64 time) {
+    UInt64 latencyNanos = 1000000000 * latencySeconds;
+    UInt64 timeElapsed = time - gLogicalInitTime;
+    UInt64 schedTime = AudioConvertNanosToHostTime(timeElapsed + gCoreAudioInitTime);
+    /* post("%lld, %lld, %lld, %lld, %lld, %lld, %lld \n",
```

Comment:
If the commented code isn't needed beyond debugging, I prefer to have it removed... if it may prove useful later, please leave a single line comment why you think it should stay in.